### PR TITLE
Added section about escape character in readme

### PR DESCRIPTION
**Pull request by pjanouse**
### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!

- [ ] Please describe what you did

- [ ] Link to relevant GitHub issues or pull requests

- [ ] Link to relevant [Jenkins JIRA issues](https://issues.jenkins-ci.org)

- [ ] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### Description

SSIA

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -216,11 +216,15 @@ The configuration file format depends on the version of jenkins-core and install
 Documentation is generated from a live instance, as well as a JSON schema you can use to validate configuration file
 with your favourite YAML tools.
 
+## Escape character
+If your need to put in an YAML definition a character which has a dedicated meaning (like character "$" or "'"), you need to escape it by character "^".
```
Comment (timja) [inline README.md]: I don't think this bit about yaml definition is correct, it is just in relation to secrets as far as i know...

Comment (pjanouse) [reply README.md]: No, we use this approach with success in other areas like:

> tool:
>   jdk:
>     installations:
>       - name: "java1.9"
>         home: "^${NATIVE_TOOLS}^${SEP}^${JAVA_9}"

Comment (timja) [reply README.md]: thats because it thinks its a secret...

Comment (pjanouse) [reply README.md]: You are right, I've take a look into the code.

On the other hand is there any limit/implication when we use the same behavior elsewhere? I couldn't find anything, but can overslooked sth. I think this escaping is useful not only for handling a secret.

Comment (casz) [reply README.md]: @pjanouse you should be able to write `home: "$NATIVE_TOOLS$SEP$JAVA_9"` with `{}` which is the delimiter used by JCasC.

Perhaps you could send a PR that adds delimiter customization, should be possible to do from the [SelfConfigurator](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/SelfConfigurator.java) and modify the [SecretResolver](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/plugin/src/main/java/io/jenkins/plugins/casc/SecretSourceResolver.java).

**Comment by timja** [comment]
Thanks for the PR @pjanouse in this case I'm -1 and don't think the PR gives an improvement to the docs, given there's been no approval from any of the other maintainers in the last few months I'm going to close this now.

Again thanks for taking the time to submit the PR

**Pull request closed without merging**